web/client/components/charts/WidgetChart.jsx

```diff
@@ -40,6 +66,64 @@ const getClassificationColors = (classifications, colorCategories, customColorEn
     })
 );
 
+/**
+ * @param {number[]} values the values of the chart to be color-coded [123.56, 24.76, ...]
+ * @param {object[]} colorCategories mapping classifiedValues => color [{min: 123.56, max: 24.76, color: "#ff0000"}, ...]
+ * @param {boolean} customColorEnabled if the user selected a custom default color
+ * @param {*} autoColorOptions default values and colors as fall back if assigned custom color is not found
+ * @returns {string[]} Hex colors for every single value in the rangeClassifications array to be mapped to
+ * a chart trace color by plotly ["#ff0000", "#00ff00"]
+ */
+const getRangeClassificationColors = (values, colorCategories, customColorEnabled, autoColorOptions) => {
+    return values.map(item => {
+        // if for some reason (error) item is not a number fall back to default color
+        if (!isNumber(item)) {
+            return defaultColorGenerator(1, COLOR_DEFAULTS)[0];
+        }
+        // try to get value in defined ranges
+        const matchedColor = colorCategories.filter(colorCategory => {
+            return item >= colorCategory.min && item < colorCategory.max;
+        })[0];
+        return matchedColor ? matchedColor.color :
+            customColorEnabled && autoColorOptions.defaultCustomColor ? autoColorOptions.defaultCustomColor :
+                defaultColorGenerator(1, COLOR_DEFAULTS)[0];
+    });
+};
+
+/**
+ * @param {string} classificationType the type of the classification attribute
+ * @param {string[] | number[]} values the values of the chart to be color-coded
+ * ["California", "Ohio", ...] or [123.45, 14.45, ...]
+ * @param {object} autoColorOptions object defining the default custom color HSV values
+ * @param {boolean} customColorEnabled if the user selected a custom default color
+ */
```

Comment:
@offtherailz correct, following your example with string values it could be like this:

```
[1,2,3,4]  /* values */
["#aaa","#aaa", "#bbb", "#bbb] /* classificatonColors */
[{ value: someValue1,  color: #aaa }, { value: someValue2,  color: #bbb }]  /* colorCategories */
```